### PR TITLE
Creating an options bag for optional parameters

### PR DESCRIPTION
**Pull request by hamshavathimunibyraiah**
Creating an options bag for optional parameters for the getTranslationsStatus and getDocumentsStatus APIs for Java SDK specifically

### CODE REVIEW

File: specification/translation/Azure.AI.DocumentTranslation/models.tsp [hunk 1]
```diff
@@ -484,3 +486,160 @@ model StartTranslationBody {
...
+    specifying a maxpagesize preference. The server SHOULD honor this preference
+    if the specified page size is smaller than the server's default page size.
+    """)
+  @maxValue(100)
+  @minValue(1)
+  @query("maxpagesize")
+  maxpagesize?: int32 = 50;
+
+  @doc("Ids to use in filtering")
+  @clientName("documentIds")
```
Comment (catalinaperalta) [inline specification/translation/Azure.AI.DocumentTranslation/models.tsp:626]: Now that we are using an actual model, please move the renames to client.tsp and remove the client generator core import from this file.

Comment (hamshavathimunibyraiah) [reply specification/translation/Azure.AI.DocumentTranslation/models.tsp:626]: Updated

File: specification/translation/Azure.AI.DocumentTranslation/models.tsp [hunk 2]
```diff
@@ -484,3 +486,160 @@ model StartTranslationBody {
...
+
+  @doc("the sorting query for the collection (ex: 'CreatedDateTimeUtc asc','CreatedDateTimeUtc desc')")
+  @query("orderby")
+  orderby?: string[];
+}
+
+@doc("Get Documents Status options")
+model GetDocumentsStatusOptions {
+  @doc("Format - uuid.  The operation id")
+  @clientName("translationId")
```
Comment (catalinaperalta) [inline specification/translation/Azure.AI.DocumentTranslation/models.tsp:569]: Please move all instances to client.tsp

File: specification/translation/Azure.AI.DocumentTranslation/models.tsp [hunk 3]
```diff
@@ -1,11 +1,13 @@
 import "@typespec/rest";
 import "@typespec/http";
 import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-client-generator-core";
```
Comment (catalinaperalta) [inline specification/translation/Azure.AI.DocumentTranslation/models.tsp:4]: remove

File: specification/translation/Azure.AI.DocumentTranslation/models.tsp [hunk 4]
```diff
@@ -1,11 +1,13 @@
...
 
 using TypeSpec.Rest;
 using TypeSpec.Http;
 using Azure.Core;
 using TypeSpec.Versioning;
+using Azure.ClientGenerator.Core;
```
Comment (catalinaperalta) [inline specification/translation/Azure.AI.DocumentTranslation/models.tsp:10]: remove

File: specification/translation/Azure.AI.DocumentTranslation/models.tsp (hunk 2 above)
Comment (hamshavathimunibyraiah) [reply specification/translation/Azure.AI.DocumentTranslation/models.tsp:569]: Updated

File: specification/translation/Azure.AI.DocumentTranslation/models.tsp (hunk 4 above)
Comment (hamshavathimunibyraiah) [reply specification/translation/Azure.AI.DocumentTranslation/models.tsp]: Removed

**Comment by mikeharder** [comment]
Check `Pipeline for Testing New Definitions` was run by mistake, and the failure can be ignored.

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**